mirrord/agent/src/steal/http/filter.rs

```diff
@@ -22,6 +29,17 @@ impl TryFrom<&mirrord_protocol::tcp::HttpFilter> for HttpFilter {
             mirrord_protocol::tcp::HttpFilter::Path(path) => {
                 Ok(Self::Path(Regex::new(&format!("(?i){path}"))?))
             }
+            mirrord_protocol::tcp::HttpFilter::Composite { all, filters } => {
+                let all = *all;
+                let mut agent_filters = vec![];
+                for filter in filters {
+                    agent_filters.push(TryFrom::try_from(filter)?);
+                }
```

Comment:
```suggestion
                let filters = filters.into_iter().map(HttpFilter::try_from).collect::<Result<Vec<_>, _>>()?;
```
Just a suggestion

Reply:
i spent so long trying to get map to work 😆 this is much nicer